### PR TITLE
Streaming Replace and Replace+Delete Support

### PR DESCRIPTION
**Pull request by niupre**
This PR introduces support for Replace only and Replace+Delete  WikiCohere runbooks.

### CODE REVIEW

**Review by HaiyangXu — APPROVED** [review]

File: BigANNBinaryEmbeddingOnlyScearioBase.cs [hunk 1]
```diff
@@ -110,7 +110,7 @@ protected override ContainerProperties GetContainerSpec(string containerName)
             return properties;
         }
 
-        protected async Task PerformIngestion(IngestionOperationType ingestionOperationType, int startVectorId, int totalVectors)
+        protected async Task PerformIngestion(IngestionOperationType ingestionOperationType, int? startTagId, int startVectorId, int totalVectors)
```
Comment (HaiyangXu) [inline BigANNBinaryEmbeddingOnlyScearioBase.cs:113]: for replacing we also do a range replacing ?

File: BigANNBinaryEmbeddingOnlyScearioBase.cs [hunk 2]
```diff
@@ -202,11 +212,6 @@ private Task<ItemResponse<EmbeddingOnlyDocument>> CreateIngestionOperationTask(I
 
         protected async Task PerformQuery(bool isWarmup, int numQueries, int KVal, string dataPath)
         {
-            if(!isWarmup)
-            {
-                this.queryMetrics[KVal] = new ScenarioMetrics(numQueries);
```
Comment (HaiyangXu) [inline BigANNBinaryEmbeddingOnlyScearioBase.cs:207]: do we want to remove warmup

File: BigANNBinaryEmbeddingOnlyScearioBase.cs (hunk 1 above)
Comment (niupre) [reply BigANNBinaryEmbeddingOnlyScearioBase.cs:113]: Yes, range defined by tags

File: BigANNBinaryEmbeddingOnlyScearioBase.cs (hunk 2 above)
Comment (niupre) [reply BigANNBinaryEmbeddingOnlyScearioBase.cs:207]: Keeping it for now.

**Pull request merged**